framework/test/unit/modules/interoperability/mainchain/commands/sidechain_registration.spec.ts

```diff
@@ -0,0 +1,504 @@
+			expect(result.error?.message).toInclude('Certificate threshold above maximum bft weight');
+		});
+	});
+
+	describe('execute', () => {
+		const genesisBlockID = Buffer.alloc(0);
+		const newChainID = intToBuffer(2, 4);
+		const existingChainID = Buffer.alloc(4);
+		existingChainID.writeUInt32BE(1, 0);
+		const params = {
+			name: 'sidechain',
+			genesisBlockID,
+			initValidators: [
+				{
+					blsKey: Buffer.from(
+						'3c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+						'hex',
+					),
+					bftWeight: BigInt(10),
+				},
+				{
+					blsKey: Buffer.from(
+						'4c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+						'hex',
+					),
+					bftWeight: BigInt(10),
+				},
+			],
+			certificateThreshold: BigInt(10),
+		};
+		const chainAccount = {
+			name: 'sidechain',
+			networkID,
+			lastCertificate: {
+				height: 0,
+				timestamp: 0,
+				stateRoot: EMPTY_HASH,
+				validatorsHash: computeValidatorsHash(params.initValidators, params.certificateThreshold),
+			},
+		};
+		const mockGetStore = jest.fn();
+		const context = {
+			logger: jest.fn(),
+			eventQueue: jest.fn(),
+			networkIdentifier: Buffer.alloc(0),
+			header: {},
+			assets: {},
+			transaction,
+			params,
+			getAPIContext: jest.fn(),
+			getStore: mockGetStore,
+		} as any;
+		const chainSubstore = {
+			getWithSchema: jest.fn().mockResolvedValue(chainAccount),
+			setWithSchema: jest.fn(),
+			iterate: jest.fn().mockResolvedValue([{ key: existingChainID, value: {} }]),
+		};
+		const channelSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const validatorsSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const outboxRootSubstore = { setWithSchema: jest.fn() };
+		const registeredNamesSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const registeredNetworkIDsSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const sendInternal = jest.fn();
+
+		beforeEach(() => {
+			sidechainRegistrationCommand['getInteroperabilityStore'] = jest
+				.fn()
+				.mockReturnValue({ sendInternal });
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA)
+				.mockReturnValue(chainSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHANNEL_DATA)
+				.mockReturnValue(channelSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_VALIDATORS)
+				.mockReturnValue(validatorsSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_OUTBOX_ROOT)
+				.mockReturnValue(outboxRootSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_REGISTERED_NAMES)
+				.mockReturnValue(registeredNamesSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_REGISTERED_NETWORK_IDS)
+				.mockReturnValue(registeredNetworkIDsSubstore);
```

Comment:
here it's using the in memory db, so it's better to just insert real value and use them

Reply:
Same as above.